# Genome assembly and annotation of _Acropora pulchra_ from Mo’orea French Polynesia

Trinity Conn1,†*, Jill Ashey2,†, Ross Cunning1 and Hollie M. Putnam2

1Conservation Research Department, John G. Shedd Aquarium, Chicago, IL, USA and 2University of Rhode Island
Dept of Biological Sciences 120 Flagg Rd. Kingston, RI 02881, USA

*corresponding author tconn@sheddaquarium.org

Reef-building corals are integral ecosystem engineers in tropical coral reefs worldwide but are increasingly threatened by climate
change and rising ocean temperatures. Consequently, there is an urgency to identify genetic, epigenetic, and environmental
factors interact for species acclimatization and adaptation. The availability of genomic resources is essential for understanding the
biology of these organisms and informing future research needs for management and and conservation. The highly diverse coral
genus _Acropora_ boasts the largest number of high-quality coral genomes, but these remain limited to a few geographic regions and
highly studied species. Here we present the first assembly and annotation of the genome and DNA methylome of _Acropora pulchra_
from Mo’orea, French Polynesia. The genome assembly was created from a combination of long-read PacBio HiFi data, from which
DNA methylation data was also called and quantified, and additional Illumina RNASeq data for ab initio gene predictions. The work
presented here resulted in the most complete _Acropora_ genome to date, with a BUSCO completeness of 96. 6% metazoan genes.
The assembly size is 518 Mbp, with 174 scaffolds, and a scaffold N50 of 17 Mbp. Structural and functional annotation resulted in the
prediction of a total of 40,518 protein-coding genes, and 16.74% of the genome in repeats. DNA methylation in the CpG context was
14.6% and predominantly found in flanking and gene body regions (61.7%). This first reference assembly of the _A. pulchra_ genome
and DNA methylome will provide the capacity for further mechanistic studies of a common coastal coral in French Polynesia of
great relevance for restoration and improve our capacity for comparative genomics in _Acropora_ and cnidarians more broadly.

We would like to thank the UC Berkeley Gump South Pacific Research Station Station for logistical support and facility space. We thank
Ariana Huffmyer, Pierrick Harnay, and Danielle Becker for field assistance, as well as Zoe Dellaert and Danielle Becker for laboratory
assistance. We would also like to acknowledge the resources provided by the John G. Shedd Aquarium. We also thank the University of Rhode
Island High Performance Computing for their logistical support. Research was completed under permits issued by the French Polynesian
Government (Délégation à la Recherche) and the Haut-commissariat de la République en Polynésie Francaise (DTRT) (Protocole d’Accueil 2005-2023).
This work represents a contribution of the Mo’orea Coral Reef LTER Site and was supported by resources from NSF-OCE
2224354 to the Mo’orea Coral Reef LTER, as well as a generous gift from the Gordon and Betty Moore Foundation. Research was completed
under permits issued by the French Polynesian Government (Délégation à la Recherche) and the Haut-commissariat de la République en
Polynésie Francaise (DTRT) (Protocole d’Accueil 2005-2023) and CITES FR2398700017-E. As guests, we recognize and give thanks for the
land and water resources of Polynesia, in particular Mo’orea, and to the traditional owners of the land, both past and present. Māuruuru roa.
With respect to the spelling of Tahitian words, we endeavored to followed the Te Fare Vāna’a transcription system that is adhered to by a
large segment of the Tahitian community, but also recognize other community members follow the Raapoto transcription system where
the island name of Moorea is, for example, spelled without the ’eta (i.e., Moorea)

