# Genome assembly and annotation of _Acropora pulchra_ from Mo’orea French Polynesia

Trinity Conn1,†*, Jill Ashey2,†, Ross Cunning1 and Hollie M. Putnam2

1Conservation Research Department, John G. Shedd Aquarium, Chicago, IL, USA and 2University of Rhode Island
Dept of Biological Sciences 120 Flagg Rd. Kingston, RI 02881, USA

*corresponding author tconn@sheddaquarium.org

Reef-building corals are integral ecosystem engineers in tropical coral reefs worldwide but are increasingly threatened by climate
change and rising ocean temperatures. Consequently, there is an urgency to identify genetic, epigenetic, and environmental
factors interact for species acclimatization and adaptation. The availability of genomic resources is essential for understanding the
biology of these organisms and informing future research needs for management and and conservation. The highly diverse coral
genus _Acropora_ boasts the largest number of high-quality coral genomes, but these remain limited to a few geographic regions and
highly studied species. Here we present the first assembly and annotation of the genome and DNA methylome of _Acropora pulchra_
from Mo’orea, French Polynesia. The genome assembly was created from a combination of long-read PacBio HiFi data, from which
DNA methylation data was also called and quantified, and additional Illumina RNASeq data for ab initio gene predictions. The work
presented here resulted in the most complete _Acropora_ genome to date, with a BUSCO completeness of 96. 6% metazoan genes.
The assembly size is 518 Mbp, with 174 scaffolds, and a scaffold N50 of 17 Mbp. Structural and functional annotation resulted in the
prediction of a total of 40,518 protein-coding genes, and 16.74% of the genome in repeats. DNA methylation in the CpG context was
14.6% and predominantly found in flanking and gene body regions (61.7%). This first reference assembly of the _A. pulchra_ genome
and DNA methylome will provide the capacity for further mechanistic studies of a common coastal coral in French Polynesia of
great relevance for restoration and improve our capacity for comparative genomics in Acropora and cnidarians more broadly.
