---
title: "Acropora pulchra genome analysis"
author: "Jill Ashey"
date: "2024-03-01"
output: html_document
---

This file will document genome analysis steps in R for *Acropora pulchra*. Most of the analysis will be done on Andromeda. My working document with Apul genome code is [here](https://github.com/JillAshey/JillAshey_Putnam_Lab_Notebook/blob/master/_posts/2024-02-06-Apulchra-Genome-Assembly.md) and the github for this project is [here](https://github.com/hputnam/Apulchra_genome). 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
```

Read in raw read length information and calculate length information (run this code after converting Hifi read file to fasta). I am keeping the raw read length file on my local computer because it is too large to push to github; it will be stored on OSF.
```{r, echo=F}
read.table(file = "~/Desktop/PutnamLab/Apulchra_genome/rr_read_lengths.txt", 
           header = F) %>% 
  dplyr::rename("hifi_read_name" = 1, 
         "length" = 2) -> hifi_read_length
nrow(hifi_read_length) # 5,898,386 total reads
mean(hifi_read_length$length) # mean length of reads is 13,424.64
sum(hifi_read_length$length) #length sum 79,183,709,778. Will need this for the NCBI submission
```

Make histogram for read bins from raw hifi data
```{r, echo = F}
rr_length_histo <- ggplot(data = hifi_read_length, 
       aes(x = length, fill = "blue")) +
  geom_histogram(binwidth = 2000) + 
  labs(x = "Raw Read Length", y = "Count", title = "Histogram of Raw HiFi Read Lengths") + 
  scale_fill_manual(values = c("blue")) + 
  scale_y_continuous(labels = function(x) format(x, scientific = FALSE)); rr_length_histo

ggsave("../output/rr_length_histogram.png", rr_length_histo)
```




