---
title: "genome plot"
author: "Jill Ashey"
date: "2025-01-10"
output: html_document
---

Plot genome and methylation information along scaffolds using the R package [karyoploteR](https://bioconductor.org/packages/release/bioc/html/karyoploteR.html). I use the terms chromosomes and scaffolds interchangeably in this script (apologies). 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#if (!requireNamespace("BiocManager", quietly = TRUE))
#  install.packages("BiocManager")
#BiocManager::install("karyoploteR")

library(tidyverse)
library(karyoploteR)
library(circlize)  # For colorRamp2
```

Plot custom genome - test data
```{r}
custom.genome <- toGRanges(data.frame(chr=c("A", "B"), start=c(1, 1), end=c(100, 200)))

kp <- plotKaryotype(genome = custom.genome)
```

Hooray it works. 

## Read in my own data 
```{r}
chroms <- read.delim("../output/assembly/ntlink/chrom_lengths.txt", sep = "", header = F)
colnames(chroms) <- c("chr", "end")

# Remove > from chr name
chroms$chr <- gsub(">", "", chroms$chr)
```

Add the starting position (1) to all rows
```{r}
chroms$start <- 1
```

Rearrange columns and organize rows
```{r}
chroms <- chroms %>%
  dplyr::select(chr, start, end) %>%
  dplyr::arrange(desc(end)) %>%
  dplyr::slice(1:10) # take top 10 chromosomes (or however many you want to plot)
  
# Make top 10 chromosome names into a character vector 
top_10_chroms <- chroms$chr
```

Manually set 'end' - do this if you want to plot a specific number of bp. Here I am plotting the first 100,000 bp of the 10 longest scaffolds in the genome. Its difficult to visualize when the scaffolds are so long. 
```{r}
chroms$end <- 100000
```

## Read in feature data 
```{r}
exon <- read.table("~/Desktop/PutnamLab/Apulchra_genome/exons.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(exon) <- c("chr", "start", "end", "gene")
exon$Type <- "Exon"
intron <- read.table("~/Desktop/PutnamLab/Apulchra_genome/introns.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(intron) <- c("chr", "start", "end", "gene")
intron$Type <- "Intron"
inter <- read.table("~/Desktop/PutnamLab/Apulchra_genome/intergenic.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(inter) <- c("chr", "start", "end")
inter$gene <- NA
inter$Type <- "Intergenic"
```

Bind feature dfs together and set colors of feature type. 
```{r}
features <- rbind(exon, intron, inter)

# Set colors 
features$gieStain <- ifelse(features$Type == "Exon", "purple",
                     ifelse(features$Type == "Intron", "orange",
                     ifelse(features$Type == "Intergenic", "gray", NA)))

# Create a custom color table for features 
color_table <- c("purple" = "purple", "orange" = "orange", "gray" = "gray")
```
The `gieStain` is specific to this package and sets the colors. It is based on UCSC's provided cytoband information, specifically whole genome Giemsa Stain track information (see info [here](https://genome.ucsc.edu/cgi-bin/hgTables)). The plotting features of this package are expecting colors in the `gieStain` column. 

Filter feature data for chromosomes of interest and bp limits 
```{r}
# Filter data for top 10 chromosomes and first 20,000 bp
filtered_features <- features[features$chr %in% top_10_chroms & 
                              features$start <= 100000, ]
```

## Read in methylation data 

Currently commented out because it takes a while to load
```{r}
# cpg <- read.table("~/Desktop/PutnamLab/Apulchra_genome/Apul.pbmm2.combined.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
# colnames(cpg) <- c("contig", "start", "end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
# dim(cpg)
```

Filter cpg data to only include the chromosomes of interest and bp limits. 
```{r}
filtered_cpg <- cpg[cpg$contig %in% top_10_chroms & 
                              cpg$start <= 100000, ]

filtered_cpg <- cpg %>%
  filter(contig %in% unique(chroms$chr)) %>%
  filter(coverage >5)
```

Remove any instances when the read coverage is less than 5. The `mod_prob` is the modification probability score or the likelihood that the cytosine in the CpG site is methylated. Higher score = higher likelihood of methylation at that site. It is currently scored as 0 to 100. Change to be a true probability between 0 and 1 for plotting purposes.
```{r}
filtered_cpg <- filtered_cpg %>%
  filter(coverage >5) %>%
  mutate(mod_prob_prob = mod_prob/100)

# Set colors 
filtered_cpg$col <- colorRamp2(c(0, max(filtered_cpg$mod_prob_prob)), c("blue", "red"))(filtered_cpg$mod_prob_prob)
```

## Prep data for plotting 

Making GRanges objects for easier plotting. The function below takes a df object and automatically finds columns that describes genomic regions, then returns them as a GRanges object. 
```{r}
# Create GRanges objects
chroms_gr <- makeGRangesFromDataFrame(chroms, 
                                   keep.extra.columns = TRUE,
                                   seqnames.field = "chr",
                                   start.field = "start",
                                   end.field = "end")
features_gr <- makeGRangesFromDataFrame(filtered_features,
                                        keep.extra.columns = TRUE)
```

## Plot! 
```{r}
# Open a PNG device to save the plot
png("../output/methylation/top_ten_scaffolds_methylation_100000bp.png", width = 50, height = 30, units = "cm", res = 300)

# Plot
kp <- plotKaryotype(genome = chroms_gr, cytobands = filtered_features, ideogram.plotter = NULL, plot.type = 1, margins = c(60, 40))

# Add background data (if needed)
#kpDataBackground(kp, data.panel = 1)

# Add cytobands with custom colors
kpAddCytobands(kp, color.table = color_table)

# Plot mod prob as points in first panel -- methylation probability 
kpPoints(kp, chr = filtered_cpg$contig, x = filtered_cpg$start, y = filtered_cpg$mod_prob_prob, col = filtered_cpg$col, pch = 16, cex = 0.5)

# Mark CpG sites
#kpPlotMarkers(kp, data = cpg_gr, labels = NULL, line.color = "black", marker.parts = c(0.5, 0.5), r1 = 0.5)

# Add axis
kpAxis(kp, ymin = 0, ymax = 1, tick.pos = c(0, .5, 1), side = 1)

dev.off()
```

I dont think there is a way to add an x-axis (see [axis](https://bernatgel.github.io/karyoploter_tutorial//Tutorial/Axis/Axis.html) in package tutorial) or a legend...

There are a lot of cool applications for this package and genome plotting, see tutorial [here](https://bernatgel.github.io/karyoploter_tutorial/). 
