---
title: "genome plot"
author: "Jill Ashey"
date: "2025-01-10"
output: html_document
---

Plot genome and methylation information along scaffolds using the R package [karyoploteR](https://bernatgel.github.io/karyoploter_tutorial/#GettingStarted). 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("karyoploteR")

library(tidyverse)
library(karyoploteR)
library(circlize)  # For colorRamp2
```

Plot custom genome - test data
```{r}
custom.genome <- toGRanges(data.frame(chr=c("A", "B"), start=c(1, 1), end=c(100, 200)))

kp <- plotKaryotype(genome = custom.genome)
```

Read in my own data 
```{r}
chroms <- read.delim("../output/assembly/ntlink/chrom_lengths.txt", sep = "", header = F)
colnames(chroms) <- c("chr", "end")

# Remove > from chr name
chroms$chr <- gsub(">", "", chroms$chr)
```

Add the starting position (1) to all rows
```{r}
chroms$start <- 1
```

Rearrange columns and organize rows
```{r}
chroms <- chroms %>%
  dplyr::select(chr, start, end) %>%
  dplyr::arrange(desc(end)) %>%
  dplyr::slice(140:150) # take 10 chromosomes
```

Make custom genome plot
```{r}
custom <- toGRanges(chroms)

kp <- plotKaryotype(genome = custom)
```

Read in feature types
```{r}
exon <- read.table("~/Desktop/PutnamLab/Apulchra_genome/exons.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(exon) <- c("chr", "start", "end", "gene")
exon$Type <- "Exon"
intron <- read.table("~/Desktop/PutnamLab/Apulchra_genome/introns.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(intron) <- c("chr", "start", "end", "gene")
intron$Type <- "Intron"
inter <- read.table("~/Desktop/PutnamLab/Apulchra_genome/intergenic.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(inter) <- c("chr", "start", "end")
inter$gene <- NA
inter$Type <- "Intergenic"
```

Bind feature dfs together 
```{r}
features <- rbind(exon, intron, inter)

# Set colors 
features$gieStain <- ifelse(features$Type == "Exon", "purple",
                     ifelse(features$Type == "Intron", "orange",
                     ifelse(features$Type == "Intergenic", "gray", NA)))
```

```{r}
#cust_feat <- toGRanges(features)
# Create a custom color table
color_table <- c("purple" = "purple", "orange" = "orange", "gray" = "gray")

kp <- plotKaryotype(genome = custom, cytobands = features)

kp <- plotKaryotype(genome = custom, cytobands = features, ideogram.plotter = NULL, plot.type = 1)
kpDataBackground(kp, data.panel = 1)
kpAddCytobands(kp, color.table = color_table)
```

Read in methylation data (commented out because it takes a while to load)
```{r}
# cpg <- read.table("~/Desktop/PutnamLab/Apulchra_genome/Apul.pbmm2.combined.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
# colnames(cpg) <- c("contig", "start", "end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
# dim(cpg)

# Filter cpg to only include contigs present in chroms
filtered_cpg <- cpg %>%
  filter(contig %in% unique(chroms$chr)) %>%
  filter(coverage >5)

# Make mod_prob into true prob between 0 and 1
filtered_cpg <- filtered_cpg %>%
  mutate(mod_prob_prob = mod_prob/100)
```

Plot
```{r}
# Prepare color mapping for methylation probabilities based on mod_prob
filtered_cpg$col <- colorRamp2(c(0, max(filtered_cpg$mod_prob_prob)), c("blue", "red"))(filtered_cpg$mod_prob_prob)

# Convert filtered_cpg to GRanges object
filtered_cpg_gr <- toGRanges(filtered_cpg)

# Open a PNG device to save the plot
png("../output/methylation/ten_scaffolds_methylation.png", width = 100, height = 20, units = "cm", res = 300)

# Create the karyotype plot
kp <- plotKaryotype(genome = custom, cytobands = features, ideogram.plotter = NULL, plot.type = 2)

# Add background data (if needed)
kpDataBackground(kp, data.panel = 1)

# Add cytobands with custom colors
kpAddCytobands(kp, color.table = color_table)

# Determine appropriate y-axis limits based on mod_prob values
#y_limits <- c(0, max(filtered_cpg$mod_prob) * 1.1)  # Set a bit above max for better visibility

# Plot mod_prob as points in the first panel with specified y-limits
kpPoints(kp, chr=filtered_cpg$contig, x=filtered_cpg$start, y=filtered_cpg$mod_prob_prob, col=filtered_cpg$col, pch=16)

# Plot regions with colors based on mod_prob in the second panel
kpPlotRegions(kp, data=filtered_cpg_gr, col="black", data.panel=2)

# Close the graphics device
dev.off()
```

try to plot 10 longest scaffolds but only plot the first 10000bp


still in progress below

```{r}
# Filter filtered_cpg to include only the first 10,000 bp of each chromosome
filtered_cpg_limited <- filtered_cpg[filtered_cpg$start <= 10000, ]

# Ensure that the end position is also set correctly for the GRanges object
filtered_cpg_limited$end <- pmin(filtered_cpg_limited$start + 1, 10000)  # Set end to start + 1 or 10000

# Convert filtered_cpg_limited to GRanges object
filtered_cpg_gr_limited <- toGRanges(filtered_cpg_limited)

# Open a PNG device to save the plot
png("../output/methylation/ten_scaffolds_methylation_10000bp.png", width = 45, height = 20, units = "cm", res = 300)

# Create the karyotype plot
kp <- plotKaryotype(genome = custom, cytobands = features, ideogram.plotter = NULL, plot.type = 2)

# Add background data (if needed)
kpDataBackground(kp, data.panel = 1)

# Add cytobands with custom colors
kpAddCytobands(kp, color.table = color_table)

# Plot mod_prob as points in the first panel for limited range
kpPoints(kp, chr=filtered_cpg_limited$contig, x=filtered_cpg_limited$start, y=filtered_cpg_limited$mod_prob, col=filtered_cpg_limited$col, pch=16)

# Plot regions with colors based on mod_prob in the second panel for limited range
kpPlotRegions(kp, data=filtered_cpg_gr_limited, col=filtered_cpg_limited$col, data.panel=2)

# Close the graphics device
dev.off()

```







