---
title: "Methylation analysis of Apul genome"
author: "Jill Ashey"
date: "2024-09-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
library(scales)
```

## CpG motifs in genome - fuzznuc 

Read in chromosome names and lengths
```{r}
lengths <- read.table("../output/assembly/ntlink/chrom_lengths.txt")
```

Rename columns 
```{r}
lengths <- lengths %>%
  rename(name = V1, length = V2)
```

Remove > in front of chromosome names 
```{r}
lengths$name <- gsub(">", "", lengths$name)
```

Read in CpG counts for each chromosome 
```{r}
cpg_counts <- read.delim("../output/methylation/CpG_chrom_counts.txt", header = F)
```

Remove extra rows 
```{r}
cpg_counts <- cpg_counts %>%
  dplyr::filter(!str_detect(V1, "#"))
```

Separate into two columns 
```{r}
# Trim whitespace from the V1 column
cpg_counts <- cpg_counts %>%
  mutate(V1 = str_trim(V1))

# Separate the column
cpg_counts <- cpg_counts %>%
  separate(col = V1, into = c("cpg_counts", "name"), sep = " ", extra = "merge")

# Set to numeric 
cpg_counts$cpg_counts <- as.numeric(cpg_counts$cpg_counts)
```

Join dfs together and calculate proportion of CpGs on each chromosome 
```{r}
all <- lengths %>%
  full_join(cpg_counts, by = "name") %>%
  mutate(proportion = (cpg_counts/length)*100)
```

Save CpG proportion of each genome to csv 
```{r}
write.csv(all, "../output/methylation/CpG_chrom_proportion.csv")
```



## pb-CpG-tools 

```{r}
cpg <- read.table("~/Desktop/PutnamLab/Apulchra_genome/Apul.pbmm2.combined.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(cpg) <- c("contig", "start", "end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
dim(cpg)
```

Column information: 

1. Reference name - contig name from reference genome (I used the genome that I assembled)
2. Start coordinate
3. End coordinate 
4. Modification score - modification probability score or the likelihood that the cytosine in the CpG site is methylated. Higher score = higher likelihood of methylation at that site 
5. Haplotype - total, probabilities combined across all reads 
6. Coverage - number of reads covering CpG site 
7. Estimated modified site count - number of CpG sites estimated to be methylated 
8. Estimated unmodified site count - number of CpG sites estimated to be unmethylated 
9. Discretized modification probability - ratio of modified to unmodified sites, indicating the confidence that the site is methylated (ie modified) or unmethylated 

Filter so that coverage > 5 counts (ie 5x coverage)
```{r}
cpg_filt <- cpg %>%
  filter(coverage > 5)
```

Group by 'contig', and calculate statistics for 'mod_prob'
```{r}
summary_stats <- cpg_filt %>%
  group_by(contig) %>%      # Group by 'contig'
  summarise(
    mean_mod_prob = mean(mod_prob, na.rm = TRUE),   # Calculate mean of 'mod_prob'
    median_mod_prob = median(mod_prob, na.rm = TRUE), # Calculate median of 'mod_prob'
    min_mod_prob = min(mod_prob, na.rm = TRUE),     # Calculate minimum of 'mod_prob'
    max_mod_prob = max(mod_prob, na.rm = TRUE)      # Calculate maximum of 'mod_prob'
  )
```

Make methylation categories and calculate how many are present in the genome. [Trigg et al. 2021](https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13542) defined methylation categories as the following: 

- Lowly or unmethylated = <10%
- Partially  methylated = 10-50%
- Methylated = 50-100%

```{r}
cpg_filt <- cpg_filt %>%
  mutate(
    methylation_category = case_when(
      mod_prob >= 0 & mod_prob <= 10 ~ "unmethylated",        # Unmethylated (0-10 mod_prob)
      mod_prob > 10 & mod_prob < 50 ~ "partially methylated", # Partially methylated (10-50 mod_prob)
      mod_prob >= 50 & mod_prob <= 100 ~ "methylated"         # Methylated (50-100 mod_prob)
    )
  )

# Count the number of rows in each category
category_counts <- cpg_filt %>%
  group_by(methylation_category) %>%
  summarise(count = n()); category_counts

# # A tibble: 3 Ã— 2
#   methylation_category    count
#   <chr>                   <int>
# 1 methylated            2321595
# 2 partially methylated   504047
# 3 unmethylated         13102755
```

Assess the percentages of each methylation category are present across the genome 
```{r}
category_counts
sum(category_counts$count) # 15928397

# Methylated 
(2321595/15928397)*100

# Partially methylayed
(504047/15928397)*100

# Unmethylated 
(13102755/15928397)*100 
```

14.6% of the CpGs are methylated, 3.2% of the CpGs are partially methylated, and 82.3% of the CpGs are unmethylated. These results are comparable to the Mcap data from Trigg et al. 2021. For Mcap, 11.4% was methylated, 9% was partially methylated, and 79.6% was unmethylated. 

## Bedtools intersect w/ pb-tools output

### Gene intersect 

Read in gene methylation intersection information
```{r}
cpg_gene <- read.table("~/Desktop/PutnamLab/Apulchra_genome/gene_methylation.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(cpg_gene) <- c("gene_contig", "gene_start", "gene_end", "gene_name", "cpg_contig", "cpg_start", "cpg_end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
length(unique(cpg_gene$gene_name))
dim(cpg_gene)
```

There are 44853 genes that have at least one CpG site There are 8251841 CpG sites total in the gemes 

Important to note: one row does not correspond to a single gene. A single gene can have multiple rows associated with it if there are multiple CpG sites in that single gene. 

Filter so that coverage > 5 counts (ie 5x coverage)
```{r}
cpg_gene_filt <- cpg_gene %>%
  filter(coverage > 5)
length(unique(cpg_gene_filt$gene_name))
dim(cpg_gene_filt)
```

There are 44816 genes that have coverage >5 There are 8,247,540 CpG sites total in the genes that have >5x coverage.

Make methylation categories and calculate how many are present in the genome. [Trigg et al. 2021](https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13542) defined methylation categories as the following: 

- Lowly or unmethylated = <10%
- Partially  methylated = 10-50%
- Methylated = 50-100%

```{r}
cpg_gene_filt <- cpg_gene_filt %>%
  mutate(
    methylation_category = case_when(
      mod_prob >= 0 & mod_prob <= 10 ~ "unmethylated",        # Unmethylated (0-10 mod_prob)
      mod_prob > 10 & mod_prob < 50 ~ "partially methylated", # Partially methylated (10-50 mod_prob)
      mod_prob >= 50 & mod_prob <= 100 ~ "methylated"         # Methylated (50-100 mod_prob)
    )
  )
```

Summarize data and plot for exons only 
```{r}
summary_data_gene <- cpg_gene_filt %>%
  group_by(methylation_category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(methylation_category = factor(methylation_category, 
                                       levels = c("methylated", "partially methylated", "unmethylated")))
levels = c("methylated", "partially methylated", "unmethylated")

# Calculate percentages
summary_data_gene <- summary_data_gene %>%
  mutate(percentage = count / sum(count) * 100)

# Create the plot
ggplot(summary_data_gene, aes(x = "", y = percentage, fill = methylation_category)) +
  geom_bar(stat = "identity", width = 1) +
  scale_fill_manual(values = c("methylated" = "red", 
                               "partially methylated" = "yellow", 
                               "unmethylated" = "blue")) +
  #coord_polar("y", start = 0) +
  theme_void() +
  labs(fill = "Methylation Category",
       title = "Methylation Categories Across All Genes") +
  geom_text(aes(label = sprintf("%.1f%%", percentage)), 
            position = position_stack(vjust = 0.5))
# Save the plot
#ggsave("exon_methylation_categories.pdf", width = 12, height = 8)
```

### Exon intersection

Read in exon methylation intersection information
```{r}
cpg_exon <- read.table("~/Desktop/PutnamLab/Apulchra_genome/exon_methylation.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(cpg_exon) <- c("exon_contig", "exon_start", "exon_end", "exon_name", "cpg_contig", "cpg_start", "cpg_end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
length(unique(cpg_exon$exon_name))
dim(cpg_exon)
```

There are 44811 genes that have a CpG site in at least one of their exons. There are 2,322,654 CpG sites total in the exons. 

Important to note: one row does not correspond to a single exon A single exon can have multiple rows associated with it if there are multiple CpG sites in that single exon 

Filter so that coverage > 5 counts (ie 5x coverage)
```{r}
cpg_exon_filt <- cpg_exon %>%
  filter(coverage > 5)
length(unique(cpg_exon_filt$exon_name))
dim(cpg_exon_filt)
```

There are 44770 genes that have coverage >5 in their exons. There are 2320826 CpG sites total in the exons that have >5x coverage.

Make methylation categories and calculate how many are present in the genome. [Trigg et al. 2021](https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13542) defined methylation categories as the following: 

- Lowly or unmethylated = <10%
- Partially  methylated = 10-50%
- Methylated = 50-100%

```{r}
cpg_exon_filt <- cpg_exon_filt %>%
  mutate(
    methylation_category = case_when(
      mod_prob >= 0 & mod_prob <= 10 ~ "unmethylated",        # Unmethylated (0-10 mod_prob)
      mod_prob > 10 & mod_prob < 50 ~ "partially methylated", # Partially methylated (10-50 mod_prob)
      mod_prob >= 50 & mod_prob <= 100 ~ "methylated"         # Methylated (50-100 mod_prob)
    )
  )
```

Summarize data and plot for exons only 
```{r}
summary_data_exon <- cpg_exon_filt %>%
  group_by(methylation_category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(methylation_category = factor(methylation_category, 
                                       levels = c("methylated", "partially methylated", "unmethylated")))
levels = c("methylated", "partially methylated", "unmethylated")

# Calculate percentages
summary_data_exon <- summary_data_exon %>%
  mutate(percentage = count / sum(count) * 100) %>%
  mutate(type = "Exon")

# Create the plot
ggplot(summary_data_exon, aes(x = "", y = percentage, fill = methylation_category)) +
  geom_bar(stat = "identity", width = 1) +
  scale_fill_manual(values = c("methylated" = "red", 
                               "partially methylated" = "yellow", 
                               "unmethylated" = "blue")) +
  #coord_polar("y", start = 0) +
  theme_void() +
  labs(fill = "Methylation Category",
       title = "Methylation Categories Across All Exons") +
  geom_text(aes(label = sprintf("%.1f%%", percentage)), 
            position = position_stack(vjust = 0.5))
# Save the plot
#ggsave("exon_methylation_categories.pdf", width = 12, height = 8)
```

### Intron intersection

Read in intron methylation intersection information
```{r}
cpg_intron <- read.table("~/Desktop/PutnamLab/Apulchra_genome/intron_methylation.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(cpg_intron) <- c("intron_contig", "intron_start", "intron_end", "intron_name", "cpg_contig", "cpg_start", "cpg_end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
length(unique(cpg_intron$intron_name))
dim(cpg_intron)
```

There are 32019 genes that have a CpG site in at least one of their introns. There are 5,796,203 CpG sites total in the introns. 

Important to note: one row does not correspond to a single intron. A single intron can have multiple rows associated with it if there are multiple CpG sites in that single intron

Filter so that coverage > 5 counts (ie 5x coverage)
```{r}
cpg_intron_filt <- cpg_intron %>%
  filter(coverage > 5)
length(unique(cpg_intron_filt$intron_name))
dim(cpg_intron_filt)
```

There are 31991 genes that have coverage >5 in their introns There are 5793730 CpG sites total in the introns that have >5x coverage.

Make methylation categories and calculate how many are present in the genome. [Trigg et al. 2021](https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13542) defined methylation categories as the following: 

- Lowly or unmethylated = <10%
- Partially  methylated = 10-50%
- Methylated = 50-100%

```{r}
cpg_intron_filt <- cpg_intron_filt %>%
  mutate(
    methylation_category = case_when(
      mod_prob >= 0 & mod_prob <= 10 ~ "unmethylated",        # Unmethylated (0-10 mod_prob)
      mod_prob > 10 & mod_prob < 50 ~ "partially methylated", # Partially methylated (10-50 mod_prob)
      mod_prob >= 50 & mod_prob <= 100 ~ "methylated"         # Methylated (50-100 mod_prob)
    )
  )
```

Summarize data and plot for introns only 
```{r}
summary_data_intron <- cpg_intron_filt %>%
  group_by(methylation_category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(methylation_category = factor(methylation_category, 
                                       levels = c("methylated", "partially methylated", "unmethylated")))
levels = c("methylated", "partially methylated", "unmethylated")

# Calculate percentages
summary_data_intron <- summary_data_intron %>%
  mutate(percentage = count / sum(count) * 100) %>%
  mutate(type = "Intron")

# Create the plot
ggplot(summary_data_intron, aes(x = "", y = percentage, fill = methylation_category)) +
  geom_bar(stat = "identity", width = 1) +
  scale_fill_manual(values = c("methylated" = "red", 
                               "partially methylated" = "yellow", 
                               "unmethylated" = "blue")) +
  #coord_polar("y", start = 0) +
  theme_void() +
  labs(fill = "Methylation Category",
       title = "Methylation Categories Across All Introns") +
  geom_text(aes(label = sprintf("%.1f%%", percentage)), 
            position = position_stack(vjust = 0.5))
# Save the plot
#ggsave("exon_methylation_categories.pdf", width = 12, height = 8)
```

### Downstream flank intersection

Read in downstream flanks methylation intersection information
```{r}
cpg_down <- read.table("~/Desktop/PutnamLab/Apulchra_genome/downstream_flanks_methylation.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(cpg_down) <- c("down_contig", "down_start", "down_end", "down_name", "cpg_contig", "cpg_start", "cpg_end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
length(unique(cpg_down$down_name))
dim(cpg_down)
```

There are 44875 downstream flanks that have at least one CpG site. There are 1,573,014 CpG sites total in the downstream flanks. 

Important to note: one row does not correspond to a single downstream flank. A single downstream flank can have multiple rows associated with it if there are multiple CpG sites in that single downstream flank.

Filter so that coverage > 5 counts (ie 5x coverage)
```{r}
cpg_down_filt <- cpg_down %>%
  filter(coverage > 5)
length(unique(cpg_down_filt$down_name))
dim(cpg_down_filt)
```

There are 44849 downstream flanks that have coverage >5. There are 1,571,971 CpG sites total in the downstream flanks that have >5x coverage.

Make methylation categories and calculate how many are present in the genome. [Trigg et al. 2021](https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13542) defined methylation categories as the following: 

- Lowly or unmethylated = <10%
- Partially  methylated = 10-50%
- Methylated = 50-100%

```{r}
cpg_down_filt <- cpg_down_filt %>%
  mutate(
    methylation_category = case_when(
      mod_prob >= 0 & mod_prob <= 10 ~ "unmethylated",        # Unmethylated (0-10 mod_prob)
      mod_prob > 10 & mod_prob < 50 ~ "partially methylated", # Partially methylated (10-50 mod_prob)
      mod_prob >= 50 & mod_prob <= 100 ~ "methylated"         # Methylated (50-100 mod_prob)
    )
  )
```

Summarize data and plot for downstream flanks only 
```{r}
summary_data_down <- cpg_down_filt %>%
  group_by(methylation_category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(methylation_category = factor(methylation_category, 
                                       levels = c("methylated", "partially methylated", "unmethylated")))
levels = c("methylated", "partially methylated", "unmethylated")

# Calculate percentages
summary_data_down <- summary_data_down %>%
  mutate(percentage = count / sum(count) * 100) %>%
  mutate(type = "Downstream Flank")

# Create the plot
ggplot(summary_data_down, aes(x = "", y = percentage, fill = methylation_category)) +
  geom_bar(stat = "identity", width = 1) +
  scale_fill_manual(values = c("methylated" = "red", 
                               "partially methylated" = "yellow", 
                               "unmethylated" = "blue")) +
  #coord_polar("y", start = 0) +
  theme_void() +
  labs(fill = "Methylation Category",
       title = "Methylation Categories Across All Downstream Flanks") +
  geom_text(aes(label = sprintf("%.1f%%", percentage)), 
            position = position_stack(vjust = 0.5))
# Save the plot
#ggsave("exon_methylation_categories.pdf", width = 12, height = 8)
```

### Upstream flank intersection

Read in upstream flanks methylation intersection information
```{r}
cpg_up <- read.table("~/Desktop/PutnamLab/Apulchra_genome/upstream_flanks_methylation.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(cpg_up) <- c("up_contig", "up_start", "up_end", "up_name", "cpg_contig", "cpg_start", "cpg_end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
length(unique(cpg_up$up_name))
dim(cpg_up)
```

There are 44889 upstream flanks that have at least one CpG site. There are 1,565,503 CpG sites total in the upstream flanks. 

Important to note: one row does not correspond to a single upstream flank. A single upstream flank can have multiple rows associated with it if there are multiple CpG sites in that single upstream flank.

Filter so that coverage > 5 counts (ie 5x coverage)
```{r}
cpg_up_filt <- cpg_up %>%
  filter(coverage > 5)
length(unique(cpg_up_filt$up_name))
dim(cpg_up_filt)
```

There are 44855 downstream flanks that have coverage >5. There are 1,564,539 CpG sites total in the downstream flanks that have >5x coverage.

Make methylation categories and calculate how many are present in the genome. [Trigg et al. 2021](https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13542) defined methylation categories as the following: 

- Lowly or unmethylated = <10%
- Partially  methylated = 10-50%
- Methylated = 50-100%

```{r}
cpg_up_filt <- cpg_up_filt %>%
  mutate(
    methylation_category = case_when(
      mod_prob >= 0 & mod_prob <= 10 ~ "unmethylated",        # Unmethylated (0-10 mod_prob)
      mod_prob > 10 & mod_prob < 50 ~ "partially methylated", # Partially methylated (10-50 mod_prob)
      mod_prob >= 50 & mod_prob <= 100 ~ "methylated"         # Methylated (50-100 mod_prob)
    )
  )
```

Summarize data and plot for upstream flanks only 
```{r}
summary_data_up <- cpg_up_filt %>%
  group_by(methylation_category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(methylation_category = factor(methylation_category, 
                                       levels = c("methylated", "partially methylated", "unmethylated")))
levels = c("methylated", "partially methylated", "unmethylated")

# Calculate percentages
summary_data_up <- summary_data_up %>%
  mutate(percentage = count / sum(count) * 100) %>%
  mutate(type = "Upstream Flank")

# Create the plot
ggplot(summary_data_up, aes(x = "", y = percentage, fill = methylation_category)) +
  geom_bar(stat = "identity", width = 1) +
  scale_fill_manual(values = c("methylated" = "red", 
                               "partially methylated" = "yellow", 
                               "unmethylated" = "blue")) +
  #coord_polar("y", start = 0) +
  theme_void() +
  labs(fill = "Methylation Category",
       title = "Methylation Categories Across All Upstream Flanks") +
  geom_text(aes(label = sprintf("%.1f%%", percentage)), 
            position = position_stack(vjust = 0.5))
# Save the plot
#ggsave("exon_methylation_categories.pdf", width = 12, height = 8)
```

### Intergenic intersection

Read in intergenic methylation intersection information
```{r}
cpg_inter <- read.table("~/Desktop/PutnamLab/Apulchra_genome/intergenic_methylation.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(cpg_inter) <- c("inter_contig", "inter_start", "inter_end", "cpg_contig", "cpg_start", "cpg_end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
dim(cpg_inter)
```

There are 7,002,915 CpG sites total in the intergenic region. 

Filter so that coverage > 5 counts (ie 5x coverage)
```{r}
cpg_inter_filt <- cpg_inter %>%
  filter(coverage > 5)
dim(cpg_inter_filt)
```

There are 6997710 CpG sites total in the intergenic region that have >5x coverage.

Make methylation categories and calculate how many are present in the genome. [Trigg et al. 2021](https://onlinelibrary.wiley.com/doi/10.1111/1755-0998.13542) defined methylation categories as the following: 

- Lowly or unmethylated = <10%
- Partially  methylated = 10-50%
- Methylated = 50-100%

```{r}
cpg_inter_filt <- cpg_inter_filt %>%
  mutate(
    methylation_category = case_when(
      mod_prob >= 0 & mod_prob <= 10 ~ "unmethylated",        # Unmethylated (0-10 mod_prob)
      mod_prob > 10 & mod_prob < 50 ~ "partially methylated", # Partially methylated (10-50 mod_prob)
      mod_prob >= 50 & mod_prob <= 100 ~ "methylated"         # Methylated (50-100 mod_prob)
    )
  )
```

Summarize data and plot for intergenic region only 
```{r}
summary_data_inter <- cpg_inter_filt %>%
  group_by(methylation_category) %>%
  summarise(count = n(), .groups = 'drop') %>%
  mutate(methylation_category = factor(methylation_category, 
                                       levels = c("methylated", "partially methylated", "unmethylated")))
levels = c("methylated", "partially methylated", "unmethylated")

# Calculate percentages
summary_data_inter <- summary_data_inter %>%
  mutate(percentage = count / sum(count) * 100) %>%
  mutate(type = "Intergenic")

# Create the plot
ggplot(summary_data_inter, aes(x = "", y = percentage, fill = methylation_category)) +
  geom_bar(stat = "identity", width = 1) +
  scale_fill_manual(values = c("methylated" = "red", 
                               "partially methylated" = "yellow", 
                               "unmethylated" = "blue")) +
  #coord_polar("y", start = 0) +
  theme_void() +
  labs(fill = "Methylation Category",
       title = "Methylation Categories Across Intergenic region") +
  geom_text(aes(label = sprintf("%.1f%%", percentage)), 
            position = position_stack(vjust = 0.5))
# Save the plot
#ggsave("exon_methylation_categories.pdf", width = 12, height = 8)
```


## Assess methylation across introns, exons, intergenic region

Bind summary data
```{r}
summary_data_all <- rbind(summary_data_exon, summary_data_intron, summary_data_down, summary_data_up, summary_data_inter)
```

Plot percentages
```{r}
# Remove unmethylated as a category 
summary_data_meth <- summary_data_all #%>%
  #dplyr::filter(methylation_category != "unmethylated")
  
# Ensure the order of factors is correct
summary_data_meth$type <- factor(summary_data_meth$type, 
                                levels = c("Exon", "Intron", "Downstream Flank", "Upstream Flank", "Intergenic"))
summary_data_meth$methylation_category <- factor(summary_data_meth$methylation_category, levels = c("unmethylated", "methylated", "partially methylated"))

# Create the plot
plot1 <- ggplot(summary_data_meth, aes(x = type, y = percentage, fill = methylation_category)) +
    geom_bar(stat = "identity", width = 0.7, color = "black", size = 0.3, position = position_stack(reverse = T)) +
scale_fill_manual(values = c("unmethylated" = "grey", "methylated" = "#5B85B1FF", "partially methylated" = "#FC766AFF"),
    labels = c("unmethylated" = "Unmethylated", "methylated" = "Methylated", "partially methylated" = "Partially methylated")) +
  scale_y_continuous(
    labels = percent_format(scale = 1), 
    expand = c(0, 0),
    breaks = seq(0, 101, by = 10),
    limits = c(0, 101)
  ) +
  coord_cartesian(clip = "off") +
  labs(x = NULL, 
       y = "Percentage of CpG sites", 
       fill = "Methylation category") +
  theme_minimal(base_size = 15) +
  theme(
    plot.title = element_text(hjust = 0.5, face = "bold", size = 16),
    plot.caption = element_text(hjust = 0, face = "italic", size = 10),
    axis.title.y = element_text(margin = margin(r = 10)),
    axis.text = element_text(color = "black"),
    axis.text.x = element_text(face = "bold", angle = 45, hjust = 1),
    legend.position = "right",
    legend.title = element_text(face = "bold"),
    panel.grid.major.x = element_blank(),
    panel.grid.minor = element_blank(),
    plot.margin = margin(20, 20, 20, 20)
  ) +
  guides(fill = guide_legend(reverse = TRUE));plot1

ggsave("../output/methylation/methylation_feature_percentage.pdf", plot = plot1, width = 10, height = 15, dpi = 300)
ggsave("../output/methylation/methylation_feature_percentage.png", plot = plot1, width = 10, height = 15, dpi = 300)
```

Plot % of CpGs found in each feature. Here, we do not care about if the CpG is methylated or not. First, calculate total count of all CpGs across features and calculate percentage of each type.
```{r}
summary_data_sum <- summary_data_all %>%
  group_by(type) %>%
  summarise(total_count = sum(count)) %>%
  mutate(percentage = total_count / sum(total_count) * 100)
```

Plot!
```{r}
# Ensure the order of factors is correct
summary_data_sum$type <- factor(summary_data_sum$type, 
                                levels = c("Exon", "Intron", "Downstream Flank", "Upstream Flank", "Intergenic"))

cpg_plot <- ggplot(summary_data_sum, aes(x = "", y = percentage, fill = type)) +
    geom_bar(stat = "identity", width = 0.7, color = "black", size = 0.3) +
    scale_fill_brewer(palette = "Set1") +
    scale_y_continuous(
        labels = percent_format(scale = 1), 
        expand = c(0, 0),
        breaks = seq(0, 100, by = 10),
        limits = c(0, 100)
    ) +
    coord_flip() +
    labs(x = NULL, 
         y = "Percentage of CpG sites", 
         fill = "Genomic region") +
    theme_minimal(base_size = 15) +
    theme(
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        panel.grid.major.y = element_blank(),
        legend.position = "right",
        legend.title = element_text(face = "bold")
    ) +
    geom_text(aes(label = sprintf("%.1f%%", percentage)), 
              position = position_stack(vjust = 0.5),
              size = 4, color = "white", fontface = "bold");cpg_plot
```







