---
title: "Methylation analysis of Apul genome"
author: "Jill Ashey"
date: "2024-09-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(tidyverse)
```

## CpG motifs in genome - fuzznuc 

Read in chromosome names and lengths
```{r}
lengths <- read.table("../output/assembly/ntlink/chrom_lengths.txt")
```

Rename columns 
```{r}
lengths <- lengths %>%
  rename(name = V1, length = V2)
```

Remove > in front of chromosome names 
```{r}
lengths$name <- gsub(">", "", lengths$name)
```

Read in CpG counts for each chromosome 
```{r}
cpg_counts <- read.delim("../output/methylation/CpG_chrom_counts.txt", header = F)
```

Remove extra rows 
```{r}
cpg_counts <- cpg_counts %>%
  dplyr::filter(!str_detect(V1, "#"))
```

Separate into two columns 
```{r}
# Trim whitespace from the V1 column
cpg_counts <- cpg_counts %>%
  mutate(V1 = str_trim(V1))

# Separate the column
cpg_counts <- cpg_counts %>%
  separate(col = V1, into = c("cpg_counts", "name"), sep = " ", extra = "merge")

# Set to numeric 
cpg_counts$cpg_counts <- as.numeric(cpg_counts$cpg_counts)
```

Join dfs together and calculate proportion of CpGs on each chromosome 
```{r}
all <- lengths %>%
  full_join(cpg_counts, by = "name") %>%
  mutate(proportion = (cpg_counts/length)*100)
```

Save CpG proportion of each genome to csv 
```{r}
write.csv(all, "../output/methylation/CpG_chrom_proportion.csv")
```



## pb-CpG-tools 

```{r}
cpg <- read.table("~/Desktop/PutnamLab/Apulchra_genome/Apul.pbmm2.combined.bed", header = F, sep="\t",stringsAsFactors=FALSE, quote="")
colnames(cpg) <- c("contig", "start", "end", "mod_prob", "haplotype", "coverage", "est_mod_count", "est_unmod_count", "est_count_ratio")
```

Column information: 

1. Reference name - contig name from reference genome (I used the genome that I assembled)
2. Start coordinate
3. End coordinate 
4. Modification score - modification probability score or the likelihood that the cytosine in the CpG site is methylated. Higher score = higher likelihood of methylation at that site 
5. Haplotype - total, probabilities combined across all reads 
6. Coverage - number of reads covering CpG site 
7. Estimated modified site count - number of CpG sites estimated to be methylated 
8. Estimated unmodified site count - number of CpG sites estimated to be unmethylated 
9. Discretized modification probability - ratio of modified to unmodified sites, indicating the confidence that the site is methylated (ie modified) or unmethylated 

Filter so that coverage > 5 counts 
```{r}
cpg_filt <- cpg %>%
  filter(coverage > 5)
```

Group by 'contig', and calculate statistics for 'mod_prob'
```{r}
summary_stats <- cpg_filt %>%
  group_by(contig) %>%      # Group by 'contig'
  summarise(
    mean_mod_prob = mean(mod_prob, na.rm = TRUE),   # Calculate mean of 'mod_prob'
    median_mod_prob = median(mod_prob, na.rm = TRUE), # Calculate median of 'mod_prob'
    min_mod_prob = min(mod_prob, na.rm = TRUE),     # Calculate minimum of 'mod_prob'
    max_mod_prob = max(mod_prob, na.rm = TRUE)      # Calculate maximum of 'mod_prob'
  )
```

Make methylation categories and calculate how many are present in the genome 
```{r}
cpg <- cpg %>%
  mutate(
    methylation_category = case_when(
      mod_prob >= 0 & mod_prob <= 20 ~ "unmethylated",        # Unmethylated (0-20 mod_prob)
      mod_prob > 20 & mod_prob < 80 ~ "partially methylated", # Partially methylated (20-80 mod_prob)
      mod_prob >= 80 & mod_prob <= 100 ~ "methylated"         # Methylated (80-100 mod_prob)
    )
  )

# Count the number of rows in each category
category_counts <- cpg %>%
  group_by(contig, methylation_category) %>%
  summarise(count = n())
```





